pyccel/ast/literals.py

```diff
@@ -33,44 +33,93 @@ class Literal(PyccelAstNode):
 class LiteralTrue(sp_BooleanTrue, Literal):
     """Represents the python value True"""
     _dtype     = NativeBool()
-    _precision = default_precision['bool']
+    def __init__(self, precision = default_precision['bool']):
+        self._precision = precision
+
+    @property
+    def precision(self):
+        return self._precision
+
+    @precision.setter
+    def precision(self, precision):
+        """ Set precision for the true literal """
+        self._precision = precision
 
 #------------------------------------------------------------------------------
 class LiteralFalse(sp_BooleanFalse, Literal):
     """Represents the python value False"""
     _dtype     = NativeBool()
-    _precision = default_precision['bool']
+    def __init__(self,precision = default_precision['bool']):
+        self._precision = precision
+
+    @property
+    def precision(self):
+        return self._precision
+
+    @precision.setter
+    def precision(self, precision):
+        """ Set precision for the false literal """
+        self._precision = precision
 
 #------------------------------------------------------------------------------
 class LiteralInteger(sp_Integer, Literal):
     """Represents an integer literal in python"""
     _dtype     = NativeInteger()
-    _precision = default_precision['int']
     def __new__(cls, val):
```

Comment:
You need to update the new method too I think. To confirm try and pass the `precision` to an instance

Reply:
what do you mean by "update"?. i searched to what can I do but couldn't find something.

Reply:
Have you tried passing the precision to an instance?